LinearAlgebraForCAP/gap/CompilerLogic.gi

```diff
@@ -54,3 +54,25 @@ CapJitAddLogicTemplate(
         needed_packages := [ [ "MatricesForHomalg", ">= 2020.05.19" ] ],
     )
 );
+
+# SyzygiesOfRows( SyzygiesOfColumns( matrix ) ) => BasisOfRows( matrix )
```

Comment:
This is a non-trivial simplification, can you please add some reasoning explaining it? I think the documentation of `BasisOfRows` and `SyzygiesOfRows` in MatricesForHomalg is not sufficient for explaining this simplification. More generally, for example `SyzygiesOfRows` does not claim to be "minimal" in any sense, so I think already using it for `KernelEmbedding` is not covered by the documentation.

Reply:
By the specification of `SyzygiesOfRows` as `WeakKernelEmbedding` and `SyzygiesOfColumns` as `WeakCokernelProjection` the composition `SyzygiesOfRows( SyzygiesOfColumns( matrix ) )` dominates `matrix` and equivalently `BasisOfRows( matrix )`. This domination is mutual when the ring is a field. It is for example false for the ring of integers:

```gap
gap> LoadPackage( "MatricesForHomalg" );
true
gap> ZZ := HomalgRingOfIntegers( );
Z
gap> m := HomalgMatrix( [ 2 ], 1, 1, ZZ );
<A 1 x 1 matrix over an internal ring>
gap> Display( m );
[ [  2 ] ]
gap> Display( SyzygiesOfRows( SyzygiesOfColumns( m ) ) );
[ [  1 ] ]
```